spark/src/main/scala/org/apache/spark/sql/delta/DeltaColumnMapping.scala

```diff
@@ -871,8 +871,14 @@ case object NameMapping extends DeltaColumnMappingMode {
 }
 
 object DeltaColumnMappingMode {
-  def apply(name: String): DeltaColumnMappingMode = {
-    name.toLowerCase(Locale.ROOT) match {
+  def apply(columnMappingModeString: String): DeltaColumnMappingMode = {
+    val columnMappingModeLowerCaseString = try {
```

Comment:
Imo this is too much defensive programming. A cleaner and more Scalatic way of doing this is
`Option(columnMappingModeString).map(case ...).getOrElse(throw ...)`
because `Option(null)` will be `None`.